quarkchain/cluster/tests/test_shard_state.py

```diff
@@ -80,6 +81,64 @@ def test_shard_state_simple(self):
             {self.genesis_token: 2500000000000000000},
         )
 
+    def test_get_total_balance(self):
+        id_list = [Identity.create_random_identity() for _ in range(11)]
+        acc_list = [Address.create_from_identity(i, full_shard_key=0) for i in id_list]
+        random_acc_list = random.sample(range(1, len(acc_list)), len(acc_list) - 1)
+        batch_size = [1, 2, 3, 4, 5, 11]
+        env = get_test_env(
+            genesis_account=acc_list[0], genesis_minor_quarkash=100000000,
+        )
+
+        qkc_token = token_id_encode("QKC")
+
+        state = create_default_shard_state(env=env)
+        # Add a root block to have all the shards initialized
+        root_block = state.root_tip.create_block_to_append().finalize()
+        state.add_root_block(root_block)
+        for i in range(len(random_acc_list)):
+            tx = create_transfer_transaction(
+                shard_state=state,
+                key=id_list[0].get_key(),
+                from_address=acc_list[0],
+                to_address=acc_list[random_acc_list[i]],
+                value=100,
+                gas=50000,
+                transfer_token_id=qkc_token,
+                gas_price=0,
+                nonce=i,
+            )
+            i += 1
```

Comment:
any special meaning for an extra increment here?

Reply:
This is to ensure that txs can be added into one block successfully.

Reply:
wait, you are iterating a range, `i` should automatically increment, right? also, how about rename `i` as `nonce`

Reply:
Oh I see...

Reply:
met with a problem saying "Failed to add transaction, 'nonce' actual: 65, target: 0" when there are 100 txs. trying to fix it.

Reply:
also, I suggest using default gas limit 21000 because the block has its own gas limit (iirc 6mil)